lib/Basic/Platform.cpp

```diff
@@ -188,3 +190,78 @@ StringRef swift::getMajorArchitectureName(const llvm::Triple &Triple) {
     return Triple.getArchName();
   }
 }
+
+static StringRef
+getArchForAppleTargetSpecificModuleTriple(const llvm::Triple &triple) {
+  auto tripleArchName = triple.getArchName();
+
+  return llvm::StringSwitch<StringRef>(tripleArchName)
+              .Cases("arm64", "aarch64", "arm64")
+              .Case ("armv7s", "armv7s")
+              .Case ("armv7k", "armv7k")
+              .Case ("armv7", "armv7")
+              .Case ("x86_64h", "x86_64h")
+              .Cases("x86_64", "amd64", "x86_64")
+              .Cases("i386", "i486", "i586", "i686", "i786", "i886", "i986",
+                     "i386")
+              .Cases("unknown", "", "unknown")
+              .Default(tripleArchName);
```

Comment:
This catch-all would cover many of the cases above; I mention them separately because I expect that build system engineers and others who don't often work in the Swift compiler may need to read this code, and I want to be extra-clear about what's supported.

Reply:
Would you want to *not* have a default, and conditionalize calling this based on the OS instead?

Reply:
This already is effectively conditionalized on `Triple::isOSDarwin()`. I just couldn't think of a good behavior if we didn't recognize the architecture (or, in other functions, the OS or environment)—this isn't a great place to diagnose an error, there's no sensible fixed default, and other parts of the compiler will diagnose anything the compiler is truly incapable of handling.

(I suppose one answer is that we ought to centralize our many disparate notions of OS, architecture, and environment into a single source of truth, but that's not something I want to tackle in this PR.)